src/libs/TransactionUtils.ts

```diff
@@ -866,20 +866,47 @@ function compareDuplicateTransactionFields(transactionID: string): {keep: Partia
         reimbursable: ['reimbursable'],
     };
 
-    const getDifferentValues = (items: Array<OnyxEntry<Transaction>>, keys: Array<keyof Transaction>) => [
-        ...new Set(
-            items
-                .map((item) => {
-                    // Prioritize modifiedMerchant over merchant
-                    if (keys.includes('modifiedMerchant' as keyof Transaction) && keys.includes('merchant' as keyof Transaction)) {
-                        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-                        return item?.modifiedMerchant || item?.merchant;
-                    }
-                    return keys.map((key) => item?.[key]);
-                })
-                .flat(),
-        ),
-    ];
+    // Helper function thats create an array of different values for a given key in the transactions
+    function getDifferentValues(items: Array<OnyxEntry<Transaction>>, keys: Array<keyof Transaction>) {
+        return [
+            ...new Set(
+                items
+                    .map((item) => {
+                        // Prioritize modifiedMerchant over merchant
+                        if (keys.includes('modifiedMerchant' as keyof Transaction) && keys.includes('merchant' as keyof Transaction)) {
+                            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                            return getMerchant(item);
+                        }
+                        return keys.map((key) => item?.[key]);
+                    })
+                    .flat(),
+            ),
+        ];
+    }
+
+    // Helper function to check if all comments are equal
+    function areAllCommentsEqual(items: Array<OnyxEntry<Transaction>>, firstTransaction: OnyxEntry<Transaction>) {
+        return items.every((item) => lodashIsEqual(item?.comment, firstTransaction?.comment));
+    }
+
+    // Helper function to check if all comments exist
+    function doAllCommentsExist(items: Array<OnyxEntry<Transaction>>, firstTransaction: OnyxEntry<Transaction>) {
+        return items.every((item) => !!item?.comment.comment === !!firstTransaction?.comment.comment);
+    }
+
+    // Helper function to check if all fields are equal for a given key
+    function areAllFieldsEqual(items: Array<OnyxEntry<Transaction>>, keyExtractor: (item: OnyxEntry<Transaction>) => string) {
+        const firstTransaction = transactions[0];
+        return items.every((item) => keyExtractor(item) === keyExtractor(firstTransaction));
+    }
+
+    // Helper function to process changes
```

Comment:
I am not going to block on this, but you could have been a bit more specific, like process changes for what purpose? Explain why we need this since the comment just say the same as the function name